go/builder.go

```diff
@@ -540,6 +545,24 @@ func (b *Builder) Slot(slotnum int) {
 	b.vtable[slotnum] = UOffsetT(b.Offset())
 }
 
+// FinishWithFileIdentifier finalizes a buffer, pointing to the given `rootTable`.
+// as well as applys a file identifier
+func (b *Builder) FinishWithFileIdentifier(rootTable UOffsetT, fid []byte) {
+	if fid == nil || len(fid) != fileIdentifierLength {
+		panic(fmt.Sprintf(
+			"file identifier must be length %d", fileIdentifierLength))
+	}
+	// In order to add a file identifier to the flatbuffer message, we need
+	// to prepare an alignment and file identifier length
+	b.Prep(SizeInt32+fileIdentifierLength, 0)
```

Comment:
This now aligns to a multiple of 8, which is not necessary, as this needs to align to 4. So just pass either one.

Reply:
It seems that the other implementations do this as well.  For example: 
https://github.com/google/flatbuffers/blob/master/java/com/google/flatbuffers/FlatBufferBuilder.java#L858
https://github.com/google/flatbuffers/blob/master/js/flatbuffers.js#L670-L671

Also, flatcc generated verification fails without the alignment a multiple of 8.

Reply:
No, that code is not quite the same. In Java, it says, "after I write these data items, I need the alignment to be `minalign`" (which is 4, or 8 if there's any long/double fields). Your code says "after I write 0 items, align to 8" (which is usually correct, but wasteful). So it be better to copy what Java does, and use minalign.